apps/web/src/hooks/use-map-thumbnail.ts

```
@@ -0,0 +1,405 @@
+        }
+      } catch (error) {
+        console.error("[handleCapture] ❌ Error:", error);
+      } finally {
+        isCapturingRef.current = false;
+      }
+    };
+
+    // Event handlers get fresh store state and pass to handleCapture
+    const handleBeforeUnload = (e: BeforeUnloadEvent) => {
+      console.log("[handleBeforeUnload] Event fired");
+      const { markers, paths, mapChanges, resetChanges } = useMapStore((state) => state);
```

Biome (2.1.2)

[error] This hook is being called from a nested function, but all hooks must be called unconditionally from the top-level component.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)